File: backend/core/src/modules/writing-goals/writing-goals.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { ProjectAccessService } from '../../common/access/project-access.service';
import { ProjectScopedCrudService } from '../../common/access/project-scoped-crud.service';
import { WritingGoal } from '../../entities';

@Injectable()
export class WritingGoalsService extends ProjectScopedCrudService<WritingGoal> {
  constructor(
    @InjectRepository(WritingGoal)
    repo: Repository<WritingGoal>,
    projectAccessService: ProjectAccessService,
  ) {
    super(repo, projectAccessService);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/service/WritingGoalsService.java
package store.pjcloud.cowrite.core.service;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.stereotype.Service;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudService;
import store.pjcloud.cowrite.core.entity.WritingGoal;
import store.pjcloud.cowrite.core.repository.WritingGoalRepository;

@Service
public class WritingGoalsService extends ProjectScopedCrudService<WritingGoal> {
    public WritingGoalsService(WritingGoalRepository repository,
                               ProjectAccessService projectAccessService,
                               ObjectMapper objectMapper) {
        super(repository, projectAccessService, objectMapper, WritingGoal.class);
    }
}
